doc/decisions/DR025-UI-delegation-plugin-interface.md

```diff
@@ -0,0 +1,184 @@
+the `ManagerInterfaceBase`-derived core interface and a
+`ManagerUIDelegate`-derived UI delegate. That is, a single plugin
+exposed two factory functions, one for the core interface and one for
+the UI delegate.
+
+The UI delegate factory function required a pre-constructed interface as
+an argument, thus avoiding the need for a global singleton to gain
+access to the core plugin from the UI plugin. However, the host
+application would request a UI delegate object through a global
+singleton `Session` object.
+
+The `ManagerUIDelegate` was expected to override methods that are
+responsible for creating or modifying PySide widgets (again, Qt based),
+much like Katana's `BaseAssetWidgetDelegate`. Since FnAssetAPI was meant
+as a generic interface, the methods were not application-specific, being
+limited to `getWidget` and `populateUI` functions, with method arguments
+providing the context for which type of widget should be constructed.
+
+## Options considered
+
+### Option 1
+
+An entirely separate UI delegate plugin, linked to the core plugin only
+by unique identifier (Katana-style).
+
+The UI delegate plugin provides a factory function for instantiating a
+UI delegate object, which in turn is responsible for creating/modifying
+UI elements. The UI delegate is provided with its corresponding
+`Manager` instance on construction, thus avoiding the need for
+singletons (FnAssetAPI-style).
+
+The UI plugin system uses the same structure as the core plugin systems.
+In particular, both C++ and Python plugins can be supported.
+
+#### Pros
+
+- Trivial to choose a different language (C++/Python) for the UI plugin
+  vs. core plugin.
+- Can add Python/C++ support first then add C++/Python support later.
+- UI delegation functionality is invisible and unobtrusive if unused.
+- Allows independent development, versioning and deployment of UI
+  plugins and core plugins. A single core plugin may work across all
+  applications, whereas a UI plugin may have a different variant for
+  each framework (i.e. Qt vs. GTK vs. wxWidgets vs. FLTK ...) or even
+  each application.
+- Very few breaking changes required, if any (perhaps the UI delegate
+  requires access to the private `ManagerInterface`?).
+
+#### Cons
+
+- Extra boilerplate code for host applications and manager plugin
+  authors.
+- Additional deployment effort for facilities maintaining yet another
+  plugin.
+- Core and UI plugins maintained and versioned independently must
+  be kept compatible and deployed in compatible pairs.
+
+
+### Option 2
+
+A single logical plugin using the existing plugin system, with
+additional method(s) added to the existing `ManagerInterface` class.
+I.e. a `getUIDelegate` (or similar) method returns a UI delegate object,
+which is responsible for creating/modifying UI elements.
+
+#### Pros
+
+- Choosing a different language (C++/Python) for the UI vs. core methods
+  is possible using the Hybrid Plugin System with the addition of a new
```

Comment:
Is the hybrid plugin system argument enough to prefer this option, as a '1st class feature' of OpenAssetIO?

Or would you rather let the hybrid system settle in a bit?

Reply:
It allows mixing and matching Python and C++, but is a bit more involved and tightly coupled compared to the other solution of having an entirely different plugin.  I guess I should expand on this.

Reply:
I was just thinking about it, because I imagine you're going to shout loudly about the Hybrid Plugin system to users, and then if it's mentioned here, but ultimately not used, that would raise an eyebrow for me. I get that it might not be the best fit, but the expansion you mention will probably cover that to make it plain for other readers.

Reply:
Expanding a bit more on the drawbacks of the hybrid plugin system for this particular use case as part of 071bb98e